src/Storages/RabbitMQ/RabbitMQConsumer.cpp

```diff
@@ -128,6 +128,32 @@ bool RabbitMQConsumer::ackMessages(const CommitInfo & commit_info)
     return false;
 }
 
+bool RabbitMQConsumer::nackMessages(const CommitInfo & commit_info)
+{
+    if (state != State::OK)
+        return false;
+
+    /// Nothing to nack.
+    if (!commit_info.delivery_tag || commit_info.delivery_tag <= last_commited_delivery_tag)
+        return false;
+
+    if (consumer_channel->reject(commit_info.delivery_tag, AMQP::multiple))
```

Comment:
By briefly searching for the AMQP API, it distinguishes between `nack` and `reject`: https://www.rabbitmq.com/nack.html

> Unfortunately, `basic.reject` provides no support for negatively acknowledging messages in bulk.
To solve this, RabbitMQ supports the `basic.nack` method that provides all the functionality of basic.reject whilst also allowing for bulk processing of messages.
>
The API also provides a `nack` flag:
https://github.com/CopernicaMarketingSoftware/AMQP-CPP/blob/master/include/amqpcpp/flags.h#L31

I wonder if it's possible to use the AMQP API? Probably that's beyond the scope of the fix.

Reply:
> Unfortunately, basic.reject provides no support for negatively acknowledging messages in bulk.

in AMQP-CPP the method `reject` has `multiple` flag so it should support bulk nacking. As far as I see in this doc https://www.rabbitmq.com/nack.html this is the only difference between `nack` and `reject`. Also near `reject` method in AMQP-CPP there is a comment "Reject or nack a message"(https://github.com/ClickHouse/AMQP-CPP/blob/00f09897ce020a84e38f87dc416af4a19c5da9ae/include/amqpcpp/channel.h#L554-L556) so should be doing both. According to https://github.com/ClickHouse/AMQP-CPP/blob/00f09897ce020a84e38f87dc416af4a19c5da9ae/src/channelimpl.cpp#L707-L713 the `reject` method does exactly what we need (sends `BasicNackFrame`). While `noAck` flag from https://github.com/ClickHouse/AMQP-CPP/blob/00f09897ce020a84e38f87dc416af4a19c5da9ae/src/flags.cpp#L27 has a comment "if set, consumed messages do not have to be acked" which means a bit different thing and is used in scope of `consume` method.
So seems using `reject` here is what we need.